sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -43,50 +31,60 @@
     prep_if_none_match,
 )
 from ._sync_token import SyncTokenPolicy
-from ._user_agent import USER_AGENT
 
 
 class AzureAppConfigurationClient:
     """Represents a client that calls restful API of Azure App Configuration service.
 
     :param str base_url: Base url of the service.
     :param credential: An object which can provide secrets for the app configuration service
-    :type credential: ~azure.appconfiguration.AppConfigConnectionStringCredential
-        or ~azure.core.credentials.TokenCredential
-    :keyword api_version: Api Version. Default value is "2022-11-01-preview". Note that overriding this default
+    :type credential: ~azure.core.credentials.TokenCredential or ~azure.core.credentials.AzureKeyCredential
+    :keyword api_version: Api Version. Default value is "2023-10-01". Note that overriding this default
         value may result in unsupported behavior.
     :paramtype api_version: str
 
     """
 
     # pylint:disable=protected-access
-    def __init__(self, base_url: str, credential: TokenCredential, **kwargs: Any) -> None:
+    def __init__(self, base_url: str, credential: Union[TokenCredential, AzureKeyCredential], **kwargs: Any) -> None:
         try:
             if not base_url.lower().startswith("http"):
-                base_url = "https://" + base_url
+                base_url = f"https://{base_url}"
         except AttributeError as exc:
             raise ValueError("Base URL must be a string.") from exc
 
         if not credential:
             raise ValueError("Missing credential")
 
-        self._credential_scopes = base_url.strip("/") + "/.default"
-
-        self._config = AzureAppConfigurationConfiguration(base_url, credential_scopes=self._credential_scopes, **kwargs)
-        self._config.user_agent_policy = UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
+        credential_scopes = [f"{base_url.strip('/')}/.default"]
         self._sync_token_policy = SyncTokenPolicy()
 
-        pipeline = kwargs.get("pipeline")
-
-        if pipeline is None:
-            aad_mode = not isinstance(credential, AppConfigConnectionStringCredential)
-            pipeline = self._create_appconfig_pipeline(
-                credential=credential, credential_scopes=self._credential_scopes, aad_mode=aad_mode, **kwargs
+        if isinstance(credential, AppConfigConnectionStringCredential):
+            self._impl = AzureAppConfiguration(
+                credential,  # type: ignore
+                base_url,
+                credential_scopes=credential_scopes,
+                per_call_policies=self._sync_token_policy,
+                authentication_policy=AppConfigRequestsCredentialsPolicy(credential),
```

Comment:
You don't need separate calls, you can just build up your kwargs differently

```python
my_kwargs = {}
if isinstance(credential, AppConfigConnectionStringCredential):
  my_kwargs.update({credential_scopes: credential_scopes, per_call_policies: self._sync_token_policy})
...
self._client = AzureAppConfiguration(credential, base_url, **my_kwargs)
```

Reply:
Updated